impeller/renderer/backend/gles/capabilities_gles.cc

```diff
@@ -106,6 +106,21 @@ CapabilitiesGLES::CapabilitiesGLES(const ProcTableGLES& gl) {
       gl.GetDescription()->HasExtension(kOESTextureBorderClampExt)) {
     supports_decal_sampler_address_mode_ = true;
   }
+
+  if (gl.GetDescription()->HasExtension(
+          "GL_EXT_multisampled_render_to_texture") &&
+      // The current implementation of MSAA support in Impeller GLES requires
+      // the use of glBlitFramebuffer, which is not available on all GLES
+      // implementations. We can't use MSAA on these platforms yet.
+      gl.BlitFramebuffer.IsAvailable()) {
+    supports_offscreen_msaa_ = true;
+
+    // We hard-code 4x MSAA, so let's make sure it's supported.
+    GLint value = 0;
+    gl.GetIntegerv(GL_MAX_SAMPLES_EXT, &value);
+    FML_DCHECK(value == 4) << "Unexpected max samples value: " << value << ". "
```

Comment:
I still reading up on the extensions spec but this check seems dodgy. We want to use 4 even when the max is higher. [Now per the dashboard](https://opengles.gpuinfo.org/displaycapability.php?name=GL_MAX_SAMPLES&esversion=3), ~83% support the 4 max. But the rest support higher numbers. In those cases, we still need to use 4.

<img width="553" alt="Screenshot 2023-10-23 at 2 53 26 PM" src="https://github.com/flutter/engine/assets/44085/fd1bddae-e157-4c1b-9266-6e4353ed8f1f">

Reply:
I suggest something like `supports_offscreen_msaa_ = value >= 4`.